data/data/powervs/cluster/iaas/iaas.tf

```
@@ -2,17 +2,9 @@ data "ibm_resource_group" "group" {
   name = var.resource_group
 }
 
-resource "ibm_resource_instance" "powervs_service_instance" {
-  name              = "${var.cluster_id}-power-iaas"
-  service           = "power-iaas"
-  plan              = "power-virtual-server-group"
-  location          = var.powervs_zone
-  tags              = ["${var.cluster_id}-power-iaas", "${var.cluster_id}"]
-  resource_group_id = data.ibm_resource_group.group.id
-
-  timeouts {
-    create = "10m"
-    update = "10m"
-    delete = "10m"
-  }
+resource "ibm_pi_workspace" "powervs_service_instance" {
+  pi_name               = "${var.cluster_id}-power-iaas"
+  pi_datacenter         = var.powervs_region
```

```suggestion
  pi_datacenter         = var.powervs_zone
```
This should take zone as i/p else how can we select which DC to create the workspace eg: dal10, dal11?

---

the documentation made it seem like it should be region and the provider level vars control where it goes. Are you sure it's zone?

[0] https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/pi_workspace